recipes/poco/all/test_package/conanfile.py

```diff
@@ -6,20 +6,29 @@ class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     generators = "cmake"
 
+    @property
+    def _with_encodings(self):
+        return "enable_encodings" in self.options["pocos"] and self.options["poco"].enable_encodings == True
```

Comment:
```suggestion
        return "enable_encodings" in self.options["poco"] and self.options["poco"].enable_encodings == True
```

typo?

Reply:
Indeed. Good catch. Thanks!

Reply:
Btw, shouldn't conan throw that I tried to access an unknown dependency?

Reply:
We try not to check everything that the users write in the recipe, otherwise, the codebase of Conan will be only about it... Here `self.options` is probably a dictionary and the `[]` operator just creates the entry (Python, where are your `const`!).

I agree we need to improve it, maybe add a wrapper over `self.options` that overrides the operator and raise if the member doesn't already exist. Feel free to open the issue, I will tag it as a feature, maybe Conan 2.0 as we need to refactor those classes.